From f193e9e15122a1f877af149773eba17c345083b2 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:18 -0400
Subject: [kernel] uapi: convert u64 to __u64 in exported headers

Message-id: <1408953210-26343-11-git-send-email-jolsa@redhat.com>
Patchwork-id: 88731
O-Subject: [PATCH RHEL7.1 BZ1133083 010/322] uapi: convert u64 to __u64 in exported headers
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0d9dfc23f4d8c17365c84eb48ecca28b963ba192
Author: Mike Frysinger <vapier@gentoo.org>
Date: Thu Jan 23 15:54:11 2014 -0800

RHEL6
=====
Only perf related hunk taken.

description
===========
The u64 type is not defined in any exported kernel headers, so trying to
use it will lead to build failures.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/uapi/linux/perf_event.h b/include/uapi/linux/perf_event.h
index 83c1b36..49fbdd4 100644
--- a/include/uapi/linux/perf_event.h
+++ b/include/uapi/linux/perf_event.h
@@ -788,6 +788,6 @@ union perf_mem_data_src {
 #define PERF_MEM_TLB_SHIFT 26
 
 #define PERF_MEM_S(a, s) \
- (((u64)PERF_MEM_##a##_##s) << PERF_MEM_##a##_SHIFT)
+ (((__u64)PERF_MEM_##a##_##s) << PERF_MEM_##a##_SHIFT)
 
 #endif /* _UAPI_LINUX_PERF_EVENT_H */
-- 
1.7.1